snorkel/learning/utils.py

```diff
@@ -1,14 +1,86 @@
 import math
+import matplotlib
 import matplotlib.pyplot as plt
 import numpy as np
 import scipy.sparse as sparse
+import warnings
+
+from pandas import DataFrame
 
-import matplotlib
 matplotlib.use('Agg')
-import warnings
 warnings.filterwarnings("ignore", module="matplotlib")
 
 
+class Scorer(object):
+    """Abstract type for scorers"""
+    def __init__(self, test_candidates, test_labels, gold_candidate_set):
+        """
+        test_candidates: candidates corresponding to test_labels
+        test_labels: ground truth labels for test candidates
+        gold_candidate_set: full set of gold labeled candidates
+        """
+        self.test_candidates = test_candidates
+        self.test_labels     = test_labels
+        self.gold_cs         = gold_candidate_set
+
+    def score(self, test_marginals, train_marginals=None, b=0.5, set_unlabeled_as_neg=True, display=True):
+        raise NotImplementedError()
+
+
```

Comment:
I think we should also add the equivalent `RelationScorer` or whatever you had in mind, with `NotImplementedError()`, just so the intended design doesn't get forgotten if we forget to finish this?